File: codeup/100000578/B.cpp
#include <cstring>
#include <iostream>

using namespace std;

const char months[13][20] = {
    "",     "January", "February",  "March",   "April",    "May",     "June",
    "July", "August",  "September", "October", "November", "December"};

const char weekday[7][15] = {"Sunday",   "Monday", "Tuesday", "Wednesday",
                             "Thursday", "Friday", "Saturday"};

int main() {
  int d, M, y, w;
  char month[15];
  while (scanf("%d %s %d", &d, month, &y) != EOF) {
    for (int i = 0; i < 13; i++) {
      if (strcmp(months[i], month) == 0) {
        M = i;
        break;
      }
    }
    if (M == 1 || M == 2) M += 12, y--;

    // 基姆拉尔森计算公式
    w = (d + 2 * M + 3 * (M + 1) / 5 + y + y / 4 - y / 100 + y / 400 + 1) % 7;
    printf("%s\n", weekday[w]);
  }

  return 0;
}

File: codeup/100000578/B.py
#!/usr/bin/python3
import sys

month_dict = {"January": 1, "February": 2, "March": 3,
              "April": 4, "May": 5, "June": 6,
              "July": 7, "August": 8, "September": 9,
              "October": 10, "November": 11, "December": 12}

week_dict = {0: "Sunday", 1: "Monday", 2: "Tuesday", 3: "Wednesday",
             4: "Thursday", 5: "Friday", 6: "Saturday"}

while True:
    tmp_time = sys.stdin.readline()
    if not tmp_time:
        break

    tmp_day, tmp_month, tmp_year = tmp_time.split()

    d = int(tmp_day)
    y = int(tmp_year)
    M = month_dict[tmp_month]
    if M == 1 or M == 2:
        M += 12
        y -= 1

    # 基姆拉尔森计算公式
    w = (d + 2 * M + 3 * (M + 1) // 5 + y +
         y // 4 - y // 100 + y // 400 + 1) % 7
    print(week_dict[w])
